front/temporal/scrub_workspace/activities.ts

```diff
@@ -132,3 +141,113 @@ async function deleteDatasources(auth: Authenticator) {
     }
   }
 }
+
+async function cleanupCustomerio(auth: Authenticator) {
+  const w = auth.workspace();
+
+  if (!w) {
+    throw new Error("No workspace found");
+  }
+
+  // Fetch all the memberships for the workspace.
+  const workspaceMemberships = await MembershipResource.getLatestMemberships({
+    workspace: w,
+  });
+
+  // Fetch all the users in the workspace.
+  const userIds = workspaceMemberships.map((m) => m.userId);
+  const users = (
+    userIds.length
+      ? await User.findAll({
+          where: {
+            id: userIds,
+          },
+        })
+      : []
+  ).map(renderUserType);
+
+  // For each user, fetch all their memberships.
+  const allMembershipsByUserId = _.groupBy(
+    userIds.length
+      ? await MembershipResource.getLatestMemberships({
+          users,
+        })
+      : [],
+    (m) => m.userId.toString()
+  );
+
+  // For every membership, fetch the workspace.
+  const workspaceIds = Object.values(allMembershipsByUserId)
+    .flat()
+    .map((m) => m.workspaceId);
+  const workspaceById = _.keyBy(
+    workspaceIds.length
+      ? await Workspace.findAll({
+          where: {
+            id: workspaceIds,
+          },
+        })
+      : [],
+    (w) => w.id.toString()
+  );
+
+  // Finally, fetch all the subscriptions for the workspaces.
+  const workspaceSids = Object.values(workspaceById).map((ws) => ws.sId);
+  const subscriptionsByWorkspaceSid = workspaceSids.length
+    ? await subscriptionForWorkspaces(workspaceSids)
+    : {};
+
+  // Process the workspace users in chunks of 16.
+  const chunks = chunk(users, 16);
+
+  for (const c of chunks) {
+    await Promise.all(
```

Comment:
IMGO no need to have concurrency here.

Reply:
Why not ?

Reply:
Bad experience lately with the `Promise.all` 🥵

Reply:
I'll meet you halfway and reduce to 4 :p